File: codeforce/round719/p4.cpp
#include <iostream>

using namespace std;

int main() {
    int t;
    cin >> t;

    while(t--) {
        int n;
        int ans = 0;
        int arr[200001];
        memset(arr, 0, 200001);

        cin >> n;

        for(int i = 0; i < n; i++) {
            int v;
            cin >> v;
            if(v - i >= 0) {
                if(i > 0) ans += arr[v-i];
                arr[v-i]++;
            }
        }

        cout << ans << endl;
    }


    return 0;
}

File: codeforce/round719/p4.py
import sys 
input = sys.stdin.readline

T = int(input())

# a[j] - j, a[i] - i
for _ in range(T):
    n = int(input())
    cnt = [0 for _ in range(10**5 * 2 + 1)]
    arr = list(map(int, input().split()))

    ans = 0
    for i in range(n):
        el = arr[i] - i
        if i == 0 and el >= 0:
            cnt[el] += 1
        else:
            if el >= 0:
                ans += cnt[el]
                cnt[el] += 1
    
    print(ans)